storage/sql/coresql/testonly/checks.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testonly
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/storage"
+)
+
+// CheckLeafContents compares two log leaves to see if they match. It fails a test with
+// a descriptive message if not.
+func CheckLeafContents(leaf *trillian.LogLeaf, seq int64, rawHash, hash, data, extraData []byte, t *testing.T) {
+	if got, want := leaf.MerkleLeafHash, hash; !bytes.Equal(got, want) {
+		t.Fatalf("Wrong leaf hash in returned leaf got\n%v\nwant:\n%v", got, want)
+	}
+
+	if got, want := leaf.LeafIdentityHash, rawHash; !bytes.Equal(got, want) {
+		t.Fatalf("Wrong raw leaf hash in returned leaf got\n%v\nwant:\n%v", got, want)
+	}
+
+	if got, want := seq, leaf.LeafIndex; got != want {
+		t.Fatalf("Bad sequence number in returned leaf got: %d, want:%d", got, want)
+	}
+
+	if got, want := leaf.LeafValue, data; !bytes.Equal(got, want) {
+		t.Fatalf("Unxpected data in returned leaf. got:\n%v\nwant:\n%v", got, want)
+	}
+
+	if got, want := leaf.ExtraData, extraData; !bytes.Equal(got, want) {
+		t.Fatalf("Unxpected data in returned leaf. got:\n%v\nwant:\n%v", got, want)
+	}
+}
+
+// LeafInBatch tests if a log leaf is contained in a slice of log leaves.
+func LeafInBatch(leaf *trillian.LogLeaf, batch []*trillian.LogLeaf) bool {
+	for _, bl := range batch {
+		if bytes.Equal(bl.LeafIdentityHash, leaf.LeafIdentityHash) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// EnsureAllLeavesDistinct checks that the supplied leaves do not contain any duplicate
+// LeafIdentityHash Values.	All the leaf hashes should be distinct because the leaves were
+// created with distinct leaf data. If only we had maps with slices as keys or sets or pretty
```

Comment:
Ah, this is obviously some strange usage of the word 'fine' that I wasn't previously aware of.